src/main/scala/edu/ie3/simona/model/em/ProportionalFlexStrat.scala

```diff
@@ -0,0 +1,138 @@
+/*
+ * © 2022. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.model.em
+
+import edu.ie3.datamodel.models.input.AssetInput
+import EmModelStrat.tolerance
+import edu.ie3.simona.ontology.messages.flex.MinMaxFlexibilityMessage.ProvideMinMaxFlexOptions
+import squants.Power
+import squants.energy.Kilowatts
+
+import java.util.UUID
+
+/** Proportionally distributes flex control among connected agents, i.e. all
+  * agents contribute the same share of their offered flex options
+  */
+object ProportionalFlexStrat extends EmModelStrat {
+
+  /** Determine the power of controllable devices by proportionally distributing
+    * flexibility usage to connected devices. This means that all devices are
+    * set to use the same share of their respective flexibility to reach target
+    * power.
+    *
+    * @param modelFlexOptions
+    *   The flex options per connected agent
+    * @param target
+    *   The target power to aim for when utilizing flexibility
+    * @return
+    *   Power set points for devices, if applicable
+    */
+  override def determineFlexControl(
+      modelFlexOptions: Iterable[
+        (_ <: AssetInput, ProvideMinMaxFlexOptions)
+      ],
+      target: Power,
+  ): Iterable[(UUID, Power)] = {
+
+    // Input models are not needed here
+    val flexOptions = modelFlexOptions
+      .map { case (_, flexOptions) =>
+        flexOptions
+      }
+
+    // sum up reference, minimum and maximum power of all connected devices
+    val (totalRef, totalMin, totalMax) = flexOptions
+      .foldLeft(
+        (Kilowatts(0d), Kilowatts(0d), Kilowatts(0d))
+      ) {
+        case (
+              (sumRef, sumMin, sumMax),
+              ProvideMinMaxFlexOptions(_, addRef, addMin, addMax),
+            ) =>
+          (
+            sumRef + addRef,
+            sumMin + addMin,
+            sumMax + addMax,
```

Comment:
I could implement a `+` operator for `ProvideMinMaxFlexOptions`, but in the end we just want a tuple of `(Power, Power, Power)` which we would need to unpack again. So that would not be shorter or wiser.

What I did instead is implement `RichIterable` in `ProvideMinMaxFlexOptions` - check it out!

Reply:
Looks good to me. Thanks!